handlers/handlers.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/Daniorocket/cookit-backend/lib"
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type Handler struct {
+	Client       *mongo.Client
+	DatabaseName string
+}
+type JWT struct {
+	Token    string
+	Username string
+}
+type Pagination struct {
+	Data          interface{} `json:"data"`
+	Limit         string      `json:"limit"`
+	Page          string      `json:"page"`
+	TotalElements int64       `json:"totalElements"`
+}
+type ApiResponse struct {
+	Data   interface{} `json:"data"`
+	Status string      `json:"status"`
+	Code   int         `json:"code"`
+	Error  string      `json:"error"`
+}
+
+func Authenticate(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch mux.CurrentRoute(r).GetName() {
+		case "CreateRecipe", "ListRecipes", "Renew", "CreateCategory", "GetUserinfo": //Todo better way? Handlers which require auth
+			tkn, username, err := lib.VerifyAndReturnJWT(w, r)
+			if err != nil {
+				log.Println("Error:", err)
+				if err := CreateApiResponse(w, nil, http.StatusUnauthorized, "failed", "Failed to authorize user"); err != nil {
+					log.Println("Error create Api response:", err)
+				}
+				return
+			}
+			ctx := context.WithValue(r.Context(), "token", JWT{Token: tkn, Username: username})
+			rWithCtx := r.WithContext(ctx)
+			h.ServeHTTP(w, rWithCtx)
```

Comment:
raczej nikt nie zapisuje WithContext do zmiennej, tylko wrzuca to już w taki sposób 
```
ctx := context.WithValue(r.Context(), "token", JWT{Token: tkn, Username: username})
h.ServeHTTP(w, r.WithContext(ctx))
```